### PR TITLE
stream error logs and add more checks based on run state

### PR DESCRIPTION
**Pull request by nickzelei**
Dependent on : https://github.com/nucleuscloud/mgmt-api/pull/323

Example:

```
NUCLEUS_DEBUG_ENV=dev

Getting deployment ready:
↪Service: trello-clone
↪Environment: acme2
↪Project Directory: /Users/nick/code/external/trello-clone

Deploying your service... ⠼ ╢▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌▌░░░░░░░░░░░░░░░░░░░░░░╟ 55 %
====================
Status Message invalid pipelineresults [url], the referred results don't exist
    Task Name: 'copy-files-from-s3-bucket'
    Start Time: '2023-01-14T12:11:47-08:00'
    Complete Time: '2023-01-14T12:12:05-08:00'
    Num Steps: '2'
        Task Step: 'download-zip-from-s3'
            Terminated: Reason: 'Completed' Message: '' Started At: '2023-01-14 12:12:04 -0800 PST' Finished At: '2023-01-14 12:12:04 -0800 PST'
        Task Step: 'decompress-zip'
            Terminated: Reason: 'Completed' Message: '' Started At: '2023-01-14 12:12:05 -0800 PST' Finished At: '2023-01-14 12:12:05 -0800 PST'
====================
Status Message invalid pipelineresults [url], the referred results don't exist
    Task Name: 'build-service'
    Start Time: '2023-01-14T12:12:05-08:00'
    Complete Time: '2023-01-14T12:16:37-08:00'
    Num Steps: '3'
        Task Step: 'prepare'
            Terminated: Reason: 'Completed' Message: '' Started At: '2023-01-14 12:12:13 -0800 PST' Finished At: '2023-01-14 12:12:13 -0800 PST'
        Task Step: 'create'
            Terminated: Reason: 'Error' Message: '' Started At: '2023-01-14 12:12:13 -0800 PST' Finished At: '2023-01-14 12:16:35 -0800 PST'
        Task Step: 'results'
            Terminated: Reason: 'Error' Message: '' Started At: '2023-01-14 12:16:35 -0800 PST' Finished At: '2023-01-14 12:16:35 -0800 PST'
====================
Printing logs for Task 'build-service' at Step 'create'
====================

Warning: Not restoring or caching layer data, no cache flag specified.
===> ANALYZING
Previous image with name "911593346794.dkr.ecr.us-west-2.amazonaws.com/nucleus-acme2-svcs:acme2-trello-clone" not found
===> DETECTING
6 of 12 buildpacks participating
paketo-buildpacks/ca-certificates 3.5.0
paketo-buildpacks/node-engine     0.20.0
paketo-buildpacks/npm-install     0.10.9
paketo-buildpacks/node-module-bom 0.4.4
paketo-buildpacks/node-run-script 0.5.3
paketo-buildpacks/npm-start       0.10.4
===> RESTORING
===> BUILDING

Paketo Buildpack for CA Certificates 3.5.0
  https://github.com/paketo-buildpacks/ca-certificates
  Launch Helper: Contributing to layer
    Creating /layers/paketo-buildpacks_ca-certificates/helper/exec.d/ca-certificates-helper
Paketo Buildpack for Node Engine 0.20.0
  Resolving Node Engine version
    Candidate version sources (in priority order):
                -> ""
      <unknown> -> ""

    Selected Node Engine version (using ): 18.12.1

  Executing build process
    Installing Node Engine 18.12.1
      Completed in 2.369s

  Generating SBOM for /layers/paketo-buildpacks_node-engine/node
      Completed in 0s

  Configuring build environment
    NODE_ENV     -> "production"
    NODE_HOME    -> "/layers/paketo-buildpacks_node-engine/node"
    NODE_VERBOSE -> "false"
    SSL_CERT_DIR -> "$SSL_CERT_DIR:/etc/ssl/certs"

  Configuring launch environment
    NODE_ENV     -> "production"
    NODE_HOME    -> "/layers/paketo-buildpacks_node-engine/node"
    NODE_VERBOSE -> "false"
    SSL_CERT_DIR -> "$SSL_CERT_DIR:/etc/ssl/certs"

    Writing exec.d/0-optimize-memory
      Calculates available memory based on container limits at launch time.
      Made available in the MEMORY_AVAILABLE environment variable.

Paketo Buildpack for NPM Install 0.10.9
  Resolving installation process
    Process inputs:
      node_modules      -> "Not found"
      npm-cache         -> "Not found"
      package-lock.json -> "Found"

    Selected NPM build process: 'npm ci'

  Executing build environment install process
    Running 'npm ci --unsafe-perm --cache /layers/paketo-buildpacks_npm-install/npm-cache'
      Completed in 1m5.865s

  Configuring build environment
    NODE_ENV -> "development"
    PATH     -> "$PATH:/layers/paketo-buildpacks_npm-install/build-modules/node_modules/.bin"

  Generating SBOM for /layers/paketo-buildpacks_npm-install/build-modules
      Completed in 1m10.108s

  Executing launch environment install process
    Running 'npm prune'
      Completed in 36.199s

  Configuring launch environment
    NODE_PROJECT_PATH   -> "/workspace/source"
    NPM_CONFIG_LOGLEVEL -> "error"
    PATH                -> "$PATH:/layers/paketo-buildpacks_npm-install/launch-modules/node_modules/.bin"

  Generating SBOM for /layers/paketo-buildpacks_npm-install/launch-modules
      Completed in 1m10.463s


Paketo Buildpack for Node Module Bill of Materials Generator 0.4.4
  Resolving CycloneDX Node.js Module version
    Selected CycloneDX Node.js Module version: 3.10.4

  Executing build process
    Installing CycloneDX Node.js Module 3.10.4
      Completed in 233ms

  Configuring environment
    Appending CycloneDX Node.js Module onto PATH

  Running CycloneDX Node.js Module
    Running 'cyclonedx-bom -o bom.json'
      Completed in 4.936s

Warning: BOM table is deprecated in this buildpack api version, though it remains supported for backwards compatibility. Buildpack authors should write BOM information to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>.
Warning: BOM table is deprecated in this buildpack api version, though it remains supported for backwards compatibility. Buildpack authors should write BOM information to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>.
Warning: BOM table is deprecated in this buildpack api version, though it remains supported for backwards compatibility. Buildpack authors should write BOM information to <layer>.sbom.<ext>, launch.sbom.<ext>, or build.sbom.<ext>.
Paketo Buildpack for Node Run Script 0.5.3
  Executing build process
    Running 'npm run build'

      > trello-clone@0.1.0 build
      > react-scripts build

      node:internal/modules/cjs/loader:535
            throw e;
            ^

      Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './lib/tokenize' is not defined by "exports" in /layers/paketo-buildpacks_npm-install/build-modules/node_modules/postcss-safe-parser/node_modules/postcss/package.json
          at new NodeError (node:internal/errors:393:5)
          at throwExportsNotFound (node:internal/modules/esm/resolve:358:9)
          at packageExportsResolve (node:internal/modules/esm/resolve:668:3)
          at resolveExports (node:internal/modules/cjs/loader:529:36)
          at Module._findPath (node:internal/modules/cjs/loader:569:31)
          at Module._resolveFilename (node:internal/modules/cjs/loader:981:27)
          at Module._load (node:internal/modules/cjs/loader:841:27)
          at Module.require (node:internal/modules/cjs/loader:1061:19)
          at require (node:internal/modules/cjs/helpers:103:18)
          at Object.<anonymous> (/layers/paketo-buildpacks_npm-install/build-modules/node_modules/postcss-safe-parser/lib/safe-parser.js:1:17) {
        code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
      }

      Node.js v18.12.1
ERROR: failed to build: exit status 1
exit status 1
pipeline failed with error
```

### CODE REVIEW

File: cmd/deploy.go [hunk 1]
```diff
@@ -278,7 +285,133 @@ func deploy(
...
+	}
+	for {
+		response, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				return nil
+			}
+			return err
+		}
+		fmt.Println(response.LogLine)
```
Comment (svrana) [inline cmd/deploy.go:354]: surprised there's no stream.Close() or whatever here.. can't remember what hte api looks like, just curious.

**Comment by svrana** [comment]
:100:

File: cmd/deploy.go (hunk 1 above)
Comment (nickzelei) [reply cmd/deploy.go:354]: When the server gracefully closes the stream, the `err` object is equal to `io.EOF`

Comment (nickzelei) [reply cmd/deploy.go:354]: The client only calls close in a bidirectional stream situation.

Comment (nickzelei) [reply cmd/deploy.go:354]: image

**Pull request merged**